pkg/cp/cmp/cmp_test.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmp
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/cp"
+)
+
+func TestCMP(t *testing.T) {
+	// Creating all tmp dirs and files for testing purpose
+	dirPath := "/tmp/u-root-pkg-cmp/"
+
+	err := os.Mkdir(dirPath, 0700)
+	if err != nil {
+		t.Fatalf("Failed to create %s: %v", dirPath, err)
+	}
+	defer os.RemoveAll(dirPath)
+
+	err = os.Mkdir(dirPath+"one", 0700)
```

Comment:
This should be done with a table. Also, I doubt the names one, two, three are more critical than 1, 2, 3, so
either
for n :-= range []string{"one", "two", "three"} {
if err := os.Mkdir(filepath.Join(dirPath, n), 0700); err != nil {
t.Fatalf(...)
}
}

Also: only one RemoveAll needed, at top level; it will get all the child directories
https://go.dev/play/p/P23Rp02dLo9

	for i := '0'; i < '4'; i++ {
		fmt.Println(filepath.Join("/tmp", string(i)))
	}